apps/hash-graph/lib/graph/src/api/rest/data_type.rs

```diff
@@ -65,15 +73,31 @@ impl RoutedResource for DataTypeResource {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase", untagged)]
+enum CreateDataTypeRequest {
+    Owned(CreateOwnedDataTypeRequest),
+    #[cfg(feature = "type-fetcher")]
+    External(CreateExternalDataTypeRequest),
+}
+
 #[derive(Debug, Serialize, Deserialize, ToSchema)]
 #[serde(rename_all = "camelCase")]
-struct CreateDataTypeRequest {
+struct CreateOwnedDataTypeRequest {
     #[schema(inline)]
     schema: MaybeListOfDataType,
     owned_by_id: OwnedById,
     actor_id: RecordCreatedById,
 }
 
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+struct CreateExternalDataTypeRequest {
+    #[schema(value_type = String)]
+    schema: VersionedUrl,
```

Comment:
Not sure how I feel about this being called `schema`. Since we're using untagged anyway, couldn't we call this `data_type_id` or something? (relevant for all ontology types)

Reply:
I think calling it `xxx_type_id` is better, yes. Going to address this after rebasing (forgot that, sorry 😞 )

Reply:
[`8debcbd` (#2576)](https://github.com/hashintel/hash/pull/2576/commits/8debcbd8bf171ed2317ff3c827f97367cfcfa002)